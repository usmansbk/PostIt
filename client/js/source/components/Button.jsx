import React from 'react';

export default function Button(props) {
  return (
    <div className={props.className}>
      <button type={props.type} className='btn waves-effect waves-light blue'>
        {props.value}
        { props.children && <i className='material-icons right'>{props.children}</i> }
      </button>
    </div>
  );
}
