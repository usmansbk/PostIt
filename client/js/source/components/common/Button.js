import React from 'react';
/**
 * Button component  that accepts optional material design icon as a child text Node
 */
export default function Button(props) {
  return (
    <div className={props.className}>
      <button type={props.type} className={'btn waves-effect waves-light ' + props.color } >
        {props.value}
        { props.children && <i className='material-icons right'>{props.children}</i> }
      </button>
    </div>
  );
}
