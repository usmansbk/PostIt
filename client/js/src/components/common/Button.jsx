import React from 'react';
/**
 * Button component  that accepts optional material design icon as a child text Node
 */
export default (props) => {
	const disabled = props.disabled && 'disabled'
  return (
    <div className={props.className}>
      <button type={props.type} className={'btn waves-effect waves-light ' + props.color + ' ' + disabled }>
        {props.value}
        { props.children && <i className='material-icons right'>{props.children}</i> }
      </button>
    </div>
  );
}
