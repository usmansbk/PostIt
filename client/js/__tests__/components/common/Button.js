import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../../src/components/common/Button';

describe('Button', () => {
  it('should render', () => {
    const component = renderer.create(<Button />)
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})