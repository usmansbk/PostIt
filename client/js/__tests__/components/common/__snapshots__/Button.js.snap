// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
<div
  className={undefined}
>
  <button
    className="btn waves-effect waves-light undefined undefined"
    type={undefined}
  />
</div>
`;
